unbalanced_dataset/under_sampling/instance_hardness_threshold.py

```diff
@@ -0,0 +1,240 @@
+"""Class to perform under-sampling based on the instance hardness
+threshold."""
+from __future__ import print_function
+from __future__ import division
+
+import numpy as np
+
+from collections import Counter
+
+from scipy.stats import mode
+
+from sklearn.utils import check_X_y
+from sklearn.neighbors import NearestNeighbors
+from sklearn.cross_validation import StratifiedKFold
+
+from .under_sampler import UnderSampler
+
+
+class InstanceHardnessThreshold(UnderSampler):
+    """Class to perform under-sampling based on the instance hardness 
+    threshold.
+
+    Parameters
+    ----------
+    estimator : sklearn classifier
+        Classifier to be used in to estimate instance hardness of the samples.
+
+    threshold : float, optional (default=0.3)
+        Threshold to be used when excluding samples (0.01 to 0.99).
+
+    mode: str, optional (default='maj')
```

Comment:
I would rename `mode` by `kind_sel`. We used this keyword in the other part of the API for almost similar. I think this is in the ENN